### PR TITLE
Support SCA team assignment with config-as-code

### PR DESCRIPTION
**Pull request by NimrodGolan**
'teamForNewProjects' was changed to 'team' duo to James request.

### CODE REVIEW

**Review by cx-aviv-sevillia — COMMENTED** [review]
let's update the wiki page according to the changes

File: src/main/java/com/checkmarx/sdk/service/scanner/ScaScanner.java [hunk 1]
```diff
@@ -138,6 +138,7 @@ private ScaConfig getScaSpecificConfig(ScanParams scanParams) {
             commonClientScaConfig.setExcludeFiles(sdkScaConfig.getExcludeFiles());
             commonClientScaConfig.setUsername(scaProperties.getUsername());
             commonClientScaConfig.setPassword(scaProperties.getPassword());
+            commonClientScaConfig.setTeam(sdkScaConfig.getTeam());
```
Comment (cx-alexey-kononov) [inline src/main/java/com/checkmarx/sdk/service/scanner/ScaScanner.java]: Can we remove a Common Client reference from here?

**Comment by NimrodGolan** [comment]
> let's update the wiki page according to the changes

I did on Cx-Flow's side.

File: src/main/java/com/checkmarx/sdk/utils/scanner/client/ScaClientHelper.java [hunk 2]
```diff
@@ -623,15 +623,25 @@ public String createRiskManagementProject(String name) throws IOException {
     }
 
     private void determineProjectTeam(CreateProjectRequest request) {
-        String team = scaProperties.getTeamForNewProjects();
+        String team = config.getScaConfig().getTeam();
```
Comment (cx-alexey-kononov) [inline src/main/java/com/checkmarx/sdk/utils/scanner/client/ScaClientHelper.java]: 1. For better readability, you can use
```suggestion
        String team = StringUtils.firstNonEmpty(config.getScaConfig().getTeam(), scaProperties.getTeam());
```
And then call setTeam(request, team) or request.setAssignedTeams(null) accordingly.

2. Also I suggest adding some logging.

File: src/main/java/com/checkmarx/sdk/utils/scanner/client/ScaClientHelper.java [hunk 3]
```diff
@@ -623,15 +623,25 @@ public String createRiskManagementProject(String name) throws IOException {
...
+                setTeam(request, team);
+            } else {
+                request.setAssignedTeams(null);
+            }
         }
     }
 
+    private void setTeam(CreateProjectRequest request, String team) {
+        log.info("Assigning SCA project with team: {}", team);
+        request.setAssignedTeams(Collections.singletonList(team));
```
Comment (cx-alexey-kononov) [inline src/main/java/com/checkmarx/sdk/utils/scanner/client/ScaClientHelper.java:641]: If user provides team A here, and the SCA project already exists with team B, what will happen with the project's teams?

File: src/main/java/com/checkmarx/sdk/service/scanner/ScaScanner.java (hunk 1 above)
Comment (NimrodGolan) [reply src/main/java/com/checkmarx/sdk/service/scanner/ScaScanner.java]: Done

File: src/main/java/com/checkmarx/sdk/utils/scanner/client/ScaClientHelper.java (hunk 2 above)
Comment (NimrodGolan) [reply src/main/java/com/checkmarx/sdk/utils/scanner/client/ScaClientHelper.java]: Nice. I updated the code.

File: src/main/java/com/checkmarx/sdk/utils/scanner/client/ScaClientHelper.java (hunk 3 above)
Comment (NimrodGolan) [reply src/main/java/com/checkmarx/sdk/utils/scanner/client/ScaClientHelper.java:641]: Team assignment is relevant only on new project creation path. If project already exists team cannot be modified or changed.

Comment (cx-alexey-kononov) [reply src/main/java/com/checkmarx/sdk/utils/scanner/client/ScaClientHelper.java:641]: Ok, so the meaning of scaProperties.getTeam() is still "team for new projects" :)
If the field name must be "team", let's at least add a relevant comment about this field behavior.

Comment (NimrodGolan) [reply src/main/java/com/checkmarx/sdk/utils/scanner/client/ScaClientHelper.java:641]: I added a comment

**Review by cx-alexey-kononov — APPROVED** [review]

**Pull request merged**